### PR TITLE
Update EntitySet.concat to use Woodwork

### PR DESCRIPTION
**Pull request by tamargrey**
Adds the concat method back into entityset, and adds additional tests

### CODE REVIEW

File: featuretools/entityset/entityset.py
Comment (tamargrey) [inline featuretools/entityset/entityset.py:899]: The previous implementation sorted on just the index column even when there was no time index. `EntitySet.update_dataframe` already sorts on time index and index if there's a time index.

If there's a reason to just sort on the index after the `concat` call, happy to add it back in, but we only ever sort when there's a time index, so I just stuck with the `update_dataframe` sorting for now.

File: featuretools/entityset/entityset.py
Comment (tamargrey) [inline featuretools/entityset/entityset.py:905]: We were previously doing `self.reset_data_description()`, but that only makes sense to me with an inplace concat call, so I changed it to reset on `combined_es` that way the concatenated entityset will always have a reset data description.

File: featuretools/entityset/entityset.py
Comment (tamargrey) [inline featuretools/entityset/entityset.py:1120]: Ran into a small bug when trying call `add_last_time_indexes` on an empty DataFrame where we couldn't do inference. Definitely an edge case, but I figured it couldn't hurt to avoid type inference here.

File: featuretools/entityset/entityset.py
Comment (thehomebrewnerd) [inline featuretools/entityset/entityset.py:873]: Eventually can use `EntitySet.dataframe_type` here.

**Review by thehomebrewnerd — COMMENTED** [review]
Overall not looking too bad, but have a few comments/questions.

**Comment by thehomebrewnerd** [comment]
One other thing - you might want to coordinate with @jeff-hernandez to determine the best order for merging this based on the work he is doing to get the `woodwork-integration` branch updated with the latest changes from `main`.

File: featuretools/tests/entityset_tests/test_es.py
Comment (tamargrey) [reply featuretools/tests/entityset_tests/test_es.py:905]: The inplace concat will allow us to test concat on a koalas entityset, so I wanted to keep the test as similar to the one above as possible while also allowing it to be used for koalas. 

If you think it'd make more sense, I can rewrite this test to not rely on copying an EntitySet

Comment (thehomebrewnerd) [reply featuretools/tests/entityset_tests/test_es.py:905]: Gotcha - I think if we are going to support concat operation with Dask/Koalas we probably need to make sure we are testing those thoroughly as all of the other tests use only the `pd_es` fixture or xfail with Dask and/or Koalas.

For this particular test I think a comment would be sufficient to note why it is different than the others.

Comment (tamargrey) [reply featuretools/tests/entityset_tests/test_es.py:905]: I can change the tests to mainly use the inplace concat, that way we can test with koalas. We'll also need to stop relying on the deepcopy calls. 

I'll see if just doing a shallow copy in those cases causes problems. 

For the places where `pd_es` is being used, those are testing the sorting that is performed only on pandas DataFrames, so I can change them to use `es` and check that the dataframes don't actually get sorted.

File: featuretools/tests/entityset_tests/test_es.py
Comment (tamargrey) [reply featuretools/tests/entityset_tests/test_es.py:973]: yep good call!!!

File: featuretools/tests/entityset_tests/test_es.py
Comment (tamargrey) [reply featuretools/tests/entityset_tests/test_es.py]: Ended up changing the implementation of deepcopy so that it works for koalas DataFrames as well. Confirmed the deepcopy tests work with this new implementation. It uses `df.ww.copy()`, which I think I like better anyway as a way of retaining the woodwork typing information.

**Review by thehomebrewnerd — APPROVED** [review]
Looks good to me, but would be good to get another set of eyes on this before merging I think.

**Review by rwedge — APPROVED** [review]
LGTM

**Pull request merged**
